core/internal/clientstate/client-state.go

```diff
@@ -116,7 +126,7 @@ func New(opts ...Option) State {
 
 	s.seqState = newSeqState()
 	s.replyState = newReplyState()
-	s.requestTimerState = newRequestTimeoutState(&s.opts)
+	s.timerState = newTimeoutState(s.opts.timerProvider, requestTimeout, prepareTimeout)
```

Comment:
Thanks @sergefdrv for clarifying the topic. That is what I should have written.

Actually, I thought like as follows, but now I find that this doesn't conform to the definition of `clientState` which use the embedding.

```
type clientState struct {
	*seqState
	*replyState
	requestTimerState timerState
	prepareTimerState timerState
	...
}
```

Reply:
I might slightly turn away from the above advice, but I added two fields of type `*timerState` to `clientState` struct. This change requires me to add some interface functions like `StartRequestTimer` directly to `clientState`, because `timerState` is not embedded in it any more. This might not be perfect, so if you have better ideas, feel free to share them.

Reply:
I think this is a good compromise :+1: